In [1]:
import pandas as pd

In [4]:
df4 = pd.read_csv("db/4훈련과정분석용.csv")

In [5]:
df4.describe()

회차         모집정원       수강신청인원       실제수강인원         수료인원  \
count  2056.000000  2056.000000  2056.000000  2056.000000  2056.000000   
mean      8.976167    28.769455    27.952335    22.151751    18.681907   
std      14.300662    12.516631    30.721457    11.944125    10.807944   
min       1.000000    15.000000     4.000000     4.000000     1.000000   
25%       2.000000    25.000000    19.000000    16.000000    13.000000   
50%       4.000000    26.000000    24.000000    21.000000    18.000000   
75%       8.000000    30.000000    30.000000    25.000000    23.000000   
max      96.000000   173.000000   526.000000   140.000000   128.000000   

               총훈련비      취업률_3개월     취업인원_3개월      취업률_6개월     취업인원_6개월  ...  \
count  2.056000e+03  2056.000000  2056.000000  2056.000000  2056.000000  ...   
mean   1.125675e+07    34.623638     5.785019    55.553405     9.405156  ...   
std    5.288065e+06    22.071182     4.635995    20.685007     5.967940  ...   
min    2.442000e+05     0.000000     0.000000     0.000000     0.000000  ...   
25%    8.529300e+06    18.200000     3.000000    41.125000     5.000000  ...   
50%    9.173730e+06    30.000000     5.000000    55.600000     8.000000  ...   
75%    1.376224e+07    46.325000     8.000000    70.100000    12.000000  ...   
max    2.904000e+07   100.000000    39.000000   100.000000    60.000000  ...   

               만족도        NCS여부        총훈련일수        총훈련시간          수료율  \
count  2056.000000  2056.000000  2056.000000  2056.000000  2056.000000   
mean    863.154183     0.021887   121.926070   944.784047    83.581211   
std      72.760361     0.146351    29.929737   225.317841    12.697230   
min     352.000000     0.000000    40.000000   201.000000    15.380000   
25%     822.000000     0.000000   115.000000   880.000000    76.920000   
50%     876.000000     0.000000   120.000000   960.000000    85.710000   
75%     917.000000     0.000000   127.000000   997.000000    93.100000   
max    1000.000000     1.000000   221.000000  1640.000000   100.000000   

             수강신청률    일 평균 훈련시간       취업당 훈련비        수도권여부          취업률  
count  2056.000000  2056.000000  2.056000e+03  2056.000000  2056.000000  
mean     92.560302     7.769314  1.663481e+06     0.821498    54.057690  
std      32.464928     0.538263  1.698512e+06     0.383028    19.662742  
min       6.670000     3.000000  0.000000e+00     0.000000     0.000000  
25%      75.000000     7.800000  7.569712e+05     1.000000    40.000000  
50%      93.845000     7.980000  1.172769e+06     1.000000    53.850000  
75%     106.670000     8.000000  1.866301e+06     1.000000    66.670000  
max     426.670000    10.970000  1.786488e+07     1.000000   100.000000  

[8 rows x 23 columns]

✅ 기본 EDA 인사이트 요약 (df4)
📌 규모 및 분포
전체 행 수: 2056개

비교적 큰 규모의 훈련 과정 데이터

회차(max=96)

동일한 과정이 반복된 경우가 많음 → 일부 기관은 반복 운영 중

🧍‍♂️ 수강 인원 관련
모집정원 평균: 28.8명 / 최대 173명

대체로 소규모 훈련

수강신청인원 max=526 vs 실제수강인원 max=140

극단적으로 인기 많지만 정원이 적은 과정 있음 → 경쟁률 높은 인기 과정 존재 가능성

수료율 평균: 83.6%

전반적으로 수료율은 높은 편

📈 훈련 성과
취업률_3개월 평균: 34.6%, 6개월 평균: 55.6%

수료 6개월 후가 더 신뢰성 있는 취업 지표로 보임

취업률 max=100%, min=0%

일부 과정은 전원 미취업, 일부는 전원 취업 → 품질 편차 큰 것인지 아니면 그냥 null값인지

취업인원_6개월 평균: 9.4명

취업에 성공한 사람 자체는 적은 편 → 소수 정예 중심

취업률 std=19.7

편차 큼. 일관성 없는 과정 많음

🧾 훈련 조건
총훈련비 평균: 1,126만 원, 최대 2,900만 원

총훈련시간 평균: 944시간 / max=1640시간

풀타임 수업(하루 8시간 기준 약 5~8개월 분량)

일 평균 훈련시간 평균: 7.77시간

전일제 중심이 많음

취업당 훈련비 평균: 약 166만 원, 최대 1786만 원

과정 효율성(ROI) 분석 필요

❤️ 만족도
만족도 평균: 863 / 1000

대체로 높은 편

만족도 std=72.76 / max=1000

일부 과정은 매우 높은 만족도 → 좋은 과정 발굴 가능성

🧾 기타
NCS 적용 비율 매우 낮음 (2.2%)

대부분 NCS 미적용 과정 → 정책 기준 중심 분석에선 유의

수도권 여부 평균: 0.82

수도권 훈련기관 비율 높음

🔍 분석 포인트 제안
✅ "비용 대비 성과" 분석 필요 (취업당 훈련비, 취업률, 만족도 간 관계)

✅ "만족도 높지만 취업률 낮은 과정" vs "취업률 높지만 만족도 낮은 과정" 비교

✅ 수도권/비수도권 간 격차 분석 가능

✅ 수료율, 취업률, 만족도 중심으로 과정 군집화 가능